### PR TITLE
feat: move to native web streams

### PR DESCRIPTION
**Pull request by talentlessguy**
resolves https://github.com/ipld/js-unixfs/issues/55

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -137,5 +133,8 @@
     "tsv": "^0.2.0",
     "typescript": "^4.8.4"
   },
-  "license": "Apache-2.0 OR MIT"
+  "license": "Apache-2.0 OR MIT",
+  "engines": {
+    "node": ">=20"
```
Comment (alanshaw) [inline package.json:138]: Is this right? You haven't actually changed any runtime code.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@ipld/unixfs",
   "description": "An implementation of the UnixFS in JavaScript designed for use with multiformats",
-  "version": "2.1.2",
+  "version": "3.0.0",
```
Comment (alanshaw) [inline package.json:4]: Please revert - this will be done automatically by CI when the module is released.

**Review by alanshaw — CHANGES_REQUESTED** [review]
This looks mostly good, just a few tweaks needed! Thanks for submitting.

I think the main thing is that there's no runtime code changes here so there's no need to release a new major version and as far as I can tell, no need to specify v20 as we've not changed anything.

Note: you need to change the node versions in github actions - I suggest we use 18 and 20 now those are LTS.

File: package.json (hunk 1 above)
Comment (talentlessguy) [reply package.json:138]: I thought of adding that since web streams, as globals, got added in Node.js 20

Comment (alanshaw) [reply package.json:138]: Yes but you don't _have_ to use the globals - you can use the polyfills as before. None of the code (outside the tests) depends on those globals.

**Review by alanshaw — APPROVED** [review]

**Comment by talentlessguy** [comment]
thx for the merge!

**Pull request merged**